qiita_db/sql_connection.py

```diff
@@ -362,11 +363,12 @@ def execute_queue(self, queue):
                 if sql_args is not None:
                     for pos, arg in enumerate(sql_args):
                         # check if previous results needed and replace
-                        if isinstance(arg, str) and \
-                                arg[0] == "{" and arg[-1] == "}":
-                            result_pos = int(arg[1:-1])
-                            sql_args[pos] = results[result_pos]
-                            clear_res = True
+                        if isinstance(arg, str):
+                            result = re.search("{(\d+)}", arg)
```

Comment:
This can probably be compiled at the class level and then called from there for speed.